pkg/controller/types/types.go

```diff
@@ -55,6 +61,28 @@ type ReconciliationRequest struct {
 	DSCI      *dsciv1.DSCInitialization
 	Release   cluster.Release
 	Manifests []ManifestInfo
+
+	//
+	// TODO: unify templates and resources.
+	//
+	// Unfortunately, the kustomize APIs do not yet support a FileSystem that is
+	// backed by golang's fs.Fs so it is not simple to have a single abstraction
+	// for both the manifests types.
+	//
+	// it would be nice to have a structure like:
+	//
+	// struct {
+	//   FS  fs.FS
+	//   URI net.URL
+	// }
+	//
+	// where the URI could be something like:
+	// - kustomize:///path/to/overlay
+	// - template:///path/to/resource.tmpl.yaml
+	//
+	// and use the scheme as discriminator for the rendering engine
+	//
+	Templates []TemplateInfo
```

Comment:
If I'm not mistaken, it looks like there is difference between Manifests and Templates. Manifests belong to ReconciliationRequest due to substitution via devFlags. But Templates are statically provided and the location can be configured on Action creation.

Reply:
Not for every component, so as an example, for kserve, the inclusion of templates depends on some configuration options (serveless, servicemesh, etc)

Reply:
Ok, I looked in more details and reconsidered.

Resources are like "context" for the pipeline, output for template or kustomize renderings and input for deploy.

With dynamic template data computation #1527 Templates serve the same for preparing configuration and input for template rendering action (so, putting them together as in comment does not sound needed).

Without that, like in existing proposal #1521, every config creation action can directly call template rendering to contribute to the Resources.

May be makes sense to put a comment before that fields that it's "Request processing Context". Well, again, if I'm not mistaken.

Reply:
make sense, I'll add some more context